### ISSUE TITLE
frequency_penalty silently no-op on chat completions (penalty has zero observable effect)

### ISSUE THREAD

**Issue by raullenchai** [raullenchai/vllm-mlx#470]
## Summary
Setting `frequency_penalty: 2.0` (the OpenAI max) has zero observable effect on output diversity. A prompt designed to elicit repetition emits the same token 10× verbatim even at max penalty.

## Repro
```bash
curl -X POST http://127.0.0.1:8501/v1/chat/completions \
  -H "Authorization: Bearer SEKRET" -H "Content-Type: application/json" \
  -d '{"model":"unsloth/Qwen3.6-27B-MLX-8bit","messages":[{"role":"user","content":"Say the word banana ten times, separated by spaces."}],"frequency_penalty":2.0,"max_tokens":40}'
```
Returns `"banana banana banana banana banana banana banana banana banana banana"` regardless of penalty value. Same output at `frequency_penalty: 0` vs `2.0`.

`presence_penalty` is accepted without error but its effect was not directly tested.

## Severity
**MED** — same "param declared but not wired" family as #459/#464/#465/#468/#469. Field IS declared on `ChatCompletionRequest` (since otherwise Pydantic would 422), but the sampler doesn't apply it. Silently degrades diversity-control workflows.

## Likely scope
Check `vllm_mlx/sampling/`: penalties may be parsed into the SamplerParams object but not actually applied during token sampling. Need to verify whether `mlx-lm`'s sampler supports frequency/presence penalties at all — if not, document the gap and either reject with 400 OR add a custom sampling pass.

Surfaced by iter15 onboarding sweep (hybrid agent).

**Comment by raullenchai** [comment]
Shipping in **v0.6.74** (release in flight, will be live on PyPI + Homebrew in a few minutes).

## Root cause

`frequency_penalty` and `presence_penalty` were already plumbed end-to-end (since #357), but mlx-lm's `make_logits_processors` defaults `frequency_context_size` and `presence_context_size` to **20**. That means the penalty only saw the last 20 generated tokens — far below a typical chat response — so the penalty value you set effectively decayed to a no-op as the response grew.

OpenAI's spec is to apply these penalties over the *entire* generated sequence.

## Fix

`vllm_mlx/scheduler.py` now passes `presence_context_size=4096` + `frequency_context_size=4096` to `make_logits_processors`, so a realistic chat-length response is fully visible to the penalty.

## Empirical before / after

On `gemma-4-12b`, prompt: *"Write 80 words about the color blue, vary your vocabulary"*:

| context_size | fp=1.5 unique/total | fp=2.0 unique/total | top word |
|---|---|---|---|
| 20 (before) | 61/81 (0.753) | 62/82 (0.756) | `the` × 6-8 |
| 4096 (after) | 71/80 (0.887) | 79/83 (0.952) | `the` gone from top-3 |

So `frequency_penalty=2.0` now visibly increases vocabulary diversity from ~0.75 → ~0.95, and the previously-dominant function word `the` is no longer the runaway top hit.

## Caveats / follow-ups

1. For generations longer than 4096 tokens (deep reasoning, code gen), the penalty is still a 4096-token sliding window, not strict "entire sequence". Tracking [#690](follow-up) to size with `max_tokens` if you hit it.
2. `repetition_penalty` is **not** changed — it's a rapid-mlx extension with multiplicative rolling-window semantics distinct from OpenAI penalties. Default window stays 20.

Fixed via #510 (merged commit 9aa324e2). Thanks for the report — the repro was clean and made A/B verification straightforward.
